File: SimpleHabit/app/src/main/java/com/me/simplehabit/persistence/dao/TopicDao.java
package com.me.simplehabit.persistence.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.me.simplehabit.data.vos.TopicVO;

import java.util.List;

@Dao
public interface TopicDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void saveTopics(List<TopicVO> topicList);

    @Query("select * from topic")
    List<TopicVO> retrieveTopics();
}

File: SimpleHabit/app/src/main/java/com/me/simplehabit/persistence/dao/TopicDao.kt
package com.me.simplehabit.persistence.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

import com.me.simplehabit.data.vos.TopicVO

@Dao
interface TopicDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun saveTopics(topicList: List<TopicVO>)

    @Query("select * from topic")
    fun retrieveTopics(): LiveData<List<TopicVO>>

    @Query("select count(*) from topic")
    fun retrieveTopicsCount(): Int
}
